src/iexamine.cpp

```diff
@@ -2780,6 +2783,19 @@ void iexamine::water_source(player &p, const tripoint &examp)
     g->handle_liquid( water, nullptr, 0, &examp );
 }
 
+void iexamine::milk_source( monster *source_mon )
```

Comment:
> I'm very noob with C++ so I have absolutely not idea of what's the difference between using pointer and reference in this situation

If you don't expect the argument to be `null`, use a reference.
Pointers can be `null` (you need to check if they aren't or document that they can't be), can point to different objects (that is, you can "reseat" a pointer) and use a different syntax that accounts for those two properties.
For someone new to C++, it is safe to use references pretty much everywhere you can and only use pointers if you can't use a reference or you have a pointer in the block above and are just using that variable.

> I don't understand what you mean by "convention", is there any info regarding conventions? 

I mean just as good practice, not codified convention.
In general, objects should be grouped by their roles/types/etc. and not share them if that is not necessary. In this case, `iexamine` handles all functions with specific set of arguments - this is useful, because we can then bind those functions to specific objects and use it like:
```
terrain_type.on_examine( who, where );
```

Your examine function is an examine function, but not "examine point by person" function, hence it would fit better somewhere else.

> I feel like you wanted to say something more(by that either) and got cut short.

Oops
I wanted to say that you can either make new files or have it all in `game.cpp`. Putting new functions to `game.cpp` is a bad practice at this point (this file is huge), but if you want to limit the changes in this PR, this would be the easiest way to go.
If you don't care about limiting number of changes, create the two new files - it will be cleaner and will be useful later if you want to add more Harvest Cataclysm functions, like shearing wool, giving animals meds and so on.